feature-libs/asm/components/customer-selection/customer-selection.component.ts

```diff
@@ -117,10 +124,92 @@ export class CustomerSelectionComponent implements OnInit, OnDestroy {
 
   closeResults() {
     this.asmService.customerSearchReset();
+    this.setFocusOnInput();
   }
 
   ngOnDestroy(): void {
     this.subscription.unsubscribe();
     this.asmService.customerSearchReset();
   }
+
+  /**
+   * set focus to the first searched item
+   * @param event keyboard event
+   */
+  focusFirstItem(event: UIEvent): void {
+    event.preventDefault();
+    this.activeIndex = 0;
+    this.updateItemIndex(this.activeIndex);
+  }
+  /**
+   * set mouse cursor to the end of search text
+   * @param event keyboard event
+   */
+  setSelectionEnd(event: UIEvent): void {
+    event.preventDefault();
+    if (this.searchTerm.nativeElement.value?.length) {
+      const selectionStart = this.searchTerm.nativeElement.value.length;
+      this.searchTerm.nativeElement.selectionStart = selectionStart;
+      this.searchTerm.nativeElement.selectionEnd = selectionStart;
+    }
+  }
+  /**
+   * set focus on previous searh result item.  If no previous item then go to end of item.
+   * @param event keyboard event
+   */
+  focusPreviousChild(event: UIEvent): void {
+    event.preventDefault();
+    this.activeIndex--;
+    if (this.activeIndex < 0) {
+      this.activeIndex = this.resultItems.length - 1;
+    }
+    this.updateItemIndex(this.activeIndex);
+  }
+  /**
+   * set focus on next searh result item.  if no next item then go to the first item
+   * @param event keyboard event
+   */
+  focusNextChild(event: UIEvent): void {
+    event.preventDefault();
+    this.activeIndex++;
+    if (this.activeIndex > this.resultItems.length - 1) {
+      this.activeIndex = 0;
+    }
+    this.updateItemIndex(this.activeIndex);
+  }
+  /**
+   * set focus to input search text
+   * @param event keyboard event
+   * @param {boolean} moveOneLeft - flag to move cursor one left
+   * @param {boolean} moveBegin - flag to move cursor to front
+   */
+  focusInputText(event: UIEvent, moveOneLeft = false, moveBegin = false): void {
+    event.preventDefault();
+    this.activeIndex = -1;
+    this.searchTerm.nativeElement.focus();
+    if (this.searchTerm.nativeElement.value?.length) {
+      let selectionPos = this.searchTerm.nativeElement.value.length;
+      if (moveOneLeft) {
+        selectionPos--;
+      }
+      if (moveBegin) {
+        selectionPos = 0;
+      }
+      this.searchTerm.nativeElement.selectionStart = selectionPos;
+      this.searchTerm.nativeElement.selectionEnd = selectionPos;
+    }
+  }
+  /**
+   * set focus to selected item
+   * @param {number} selectedIndex - current selected item index
+   */
+  updateItemIndex(selectedIndex: number): void {
+    this.resultItems.toArray()?.[selectedIndex]?.nativeElement.focus();
+  }
+
+  setFocusOnInput(): void {
+    setTimeout(() => {
+      this.searchTerm.nativeElement.focus();
+    });
```

Comment:
I don't think you should use `setTimeout` - if the issue is change detection, use `ChangeDetectorRef:detectChanges`.

Reply:
without setTimeout it get focus...

Reply:
$event.preventDefault() help on Enter is clicked but not ESC is clicked so need to keep setTimeout for closing the result popup

Reply:
found a solution... 
we need both event handlings for ESC key
event.preventDefault();
event.stopPropagation();